hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/internal/DataSourceInternalWriterHelper.java

```diff
@@ -66,13 +66,11 @@ public DataSourceInternalWriterHelper(String instantTime, HoodieWriteConfig writ
     this.extraMetadata = extraMetadata;
     this.writeClient = new SparkRDDWriteClient<>(new HoodieSparkEngineContext(new JavaSparkContext(sparkSession.sparkContext())), writeConfig);
     this.writeClient.setOperationType(operationType);
-    this.writeClient.startCommitWithTime(instantTime);
     this.writeClient.initTable(operationType, Option.of(instantTime));
```

Comment:
Not sure if `initTable` requires the requested commit file to be existed.

Reply:
Definitely, initTable does not require the requested commit file to be existed. That's why i moved invocation of startCommitWithTime out of constructor right to that place, where it is needed (first statement of  createInflightCommit method).

Reply:
The `initTable` has a param `Option<String> instantTime` which is used for startting the trasanction conflict process, not sure what happens if the metadata file does not exist there at all.

Reply:
oh, yes, may be you are right, but all tests pass succesfully. Can you give me some advice to make this PR correct?

Reply:
I have no strong advices in this moment, the lifecycle methods in write client has it's own dependencies, this is really a pain to maintainence, maybe we should add more docs for them first and find out a way to make the method invocation more safe.